File: server/src/main/java/com/uiptv/server/HttpIconServer.java
package com.uiptv.server;

import com.sun.net.httpserver.HttpExchange;
import com.sun.net.httpserver.HttpHandler;
import org.apache.commons.io.IOUtils;

import java.io.IOException;
import java.io.InputStream;
import java.io.OutputStream;
import java.nio.file.Files;
import java.nio.file.Path;

public class HttpIconServer implements HttpHandler {
    private static final String RESOURCE_ICON = "/icon.ico";

    @Override
    public void handle(HttpExchange ex) throws IOException {
        byte[] bytes = null;

        try {
            Path filePath = StaticWebFileResolver.resolve(ex);
            bytes = Files.readAllBytes(filePath);
        } catch (IOException _) {
            // Fall through to classpath.
        }

        if (bytes == null) {
            try (InputStream is = HttpIconServer.class.getResourceAsStream(RESOURCE_ICON)) {
                if (is != null) {
                    bytes = IOUtils.toByteArray(is);
                }
            }
        }

        if (bytes == null) {
            ex.sendResponseHeaders(404, -1);
            return;
        }

        ex.getResponseHeaders().add("Content-Type", "image/x-icon");
        ex.sendResponseHeaders(200, bytes.length);
        try (OutputStream os = ex.getResponseBody()) {
            os.write(bytes);
        }
    }
}

File: server/src/main/kotlin/com/uiptv/server/HttpIconServer.kt
package com.uiptv.server

import com.sun.net.httpserver.HttpExchange
import com.sun.net.httpserver.HttpHandler
import org.apache.commons.io.IOUtils
import java.io.IOException
import java.nio.file.Files

class HttpIconServer : HttpHandler {
    companion object {
        private const val RESOURCE_ICON = "/icon.ico"
    }

    @Throws(IOException::class)
    override fun handle(ex: HttpExchange) {
        var bytes: ByteArray? = null
        try {
            val filePath = StaticWebFileResolver.resolve(ex)
            bytes = Files.readAllBytes(filePath)
        } catch (_: IOException) {
        }

        if (bytes == null) {
            HttpIconServer::class.java.getResourceAsStream(RESOURCE_ICON)?.use { input ->
                bytes = IOUtils.toByteArray(input)
            }
        }
        if (bytes == null) {
            ex.sendResponseHeaders(404, -1)
            return
        }
        ex.responseHeaders.add("Content-Type", "image/x-icon")
        ex.sendResponseHeaders(200, bytes!!.size.toLong())
        ex.responseBody.use { os -> os.write(bytes) }
    }
}
